mapper-pojo/src/main/java/org/hibernate/search/v6poc/entity/pojo/mapping/definition/annotation/impl/AnnotationPojoTypeNodeMetadataContributorImpl.java

```diff
@@ -0,0 +1,278 @@
+		propertyModel.getAnnotationsByMetaAnnotationType( MarkerMapping.class )
+				.forEach( annotation -> addMarker( collector.property( name ), annotation ) );
+	}
+
+	private void contributePropertyMapping(PojoTypeNodeMappingCollector collector, PropertyModel<?> propertyModel) {
+		PropertyHandle handle = propertyModel.getHandle();
+		propertyModel.getAnnotationsByType( DocumentId.class )
+				.forEach( annotation -> addDocumentId( collector.property( handle ), propertyModel, annotation ) );
+		propertyModel.getAnnotationsByMetaAnnotationType( BridgeMapping.class )
+				.forEach( annotation -> addBridge( collector.property( handle ), annotation ) );
+		propertyModel.getAnnotationsByType( Field.class )
+				.forEach( annotation -> addField( collector.property( handle ), propertyModel, annotation ) );
+		propertyModel.getAnnotationsByType( IndexedEmbedded.class )
+				.forEach( annotation -> addIndexedEmbedded( collector.property( handle ), propertyModel, annotation ) );
+	}
+
+	private <A extends Annotation> void addMarker(PojoPropertyNodeModelCollector collector, A annotation) {
+		AnnotationMarkerBuilder<A> builder = createMarkerBuilder( annotation );
+		builder.initialize( annotation );
+		collector.marker( builder );
+	}
+
+	private void addDocumentId(PojoPropertyNodeMappingCollector collector, PropertyModel<?> propertyModel, DocumentId annotation) {
+		BridgeBuilder<? extends IdentifierBridge<?>> builder = createIdentifierBridgeBuilder( annotation, propertyModel );
+
+		collector.identifierBridge( builder );
+	}
+
+	private <A extends Annotation> void addBridge(PojoNodeMappingCollector collector, A annotation) {
+		AnnotationBridgeBuilder<? extends Bridge, A> builder = createBridgeBuilder( annotation );
+		builder.initialize( annotation );
+		collector.bridge( builder );
+	}
+
+	private void addField(PojoPropertyNodeMappingCollector collector, PropertyModel<?> propertyModel, Field annotation) {
+		String cleanedUpFieldName = annotation.name();
+		if ( cleanedUpFieldName.isEmpty() ) {
+			cleanedUpFieldName = null;
+		}
+
+		BridgeBuilder<? extends FunctionBridge<?, ?>> builder = createFunctionBridgeBuilder( annotation, propertyModel );
+
+		collector.functionBridge( builder, cleanedUpFieldName, new AnnotationFieldModelContributor( annotation ) );
+	}
+
+	private void addIndexedEmbedded(PojoPropertyNodeMappingCollector collector, PropertyModel<?> propertyModel,
+			IndexedEmbedded annotation) {
+		String cleanedUpPrefix = annotation.prefix();
+		if ( cleanedUpPrefix.isEmpty() ) {
+			cleanedUpPrefix = null;
+		}
+
+		Integer cleanedUpMaxDepth = annotation.maxDepth();
+		if ( cleanedUpMaxDepth.equals( -1 ) ) {
+			cleanedUpMaxDepth = null;
+		}
+
+		String[] includePathsArray = annotation.includePaths();
+		Set<String> cleanedUpIncludePaths;
+		if ( includePathsArray.length > 0 ) {
+			cleanedUpIncludePaths = new HashSet<>();
+			Collections.addAll( cleanedUpIncludePaths, includePathsArray );
+		}
+		else {
+			cleanedUpIncludePaths = null;
```

Comment:
Is it such a good idea to use `null` here? Having an empty set seems to make more sense to me but maybe I'm missing something.